File: September 2024/September 16/kaprekar_num.py
import math

def kaprekar_numbers(p, q):
    answer = ""
    while p <= q:
        if is_kaprekar(p):
            answer = answer + p + ""
        p+=1
        
    if len(answer)>0:
        print(answer)
    else:
        print("INVALID RANGE")

def is_kaprekar(num):
    if num == 1:
        return True
    
    sq_num = str(num*num)
    length = math.floor(len(sq_num)/2)
    x = slice(0, length)
    y = slice(length)
    
    total = int(sq_num[x]) + int(sq_num[y])

    if total == num

File: September 2024/September 16/kaprekarNum.js
function kaprekarNumbers(p, q) {
    let answer = ""
  
    while(p<=q){
      if(isKaprekar(p)){
        answer = answer + p + " "
      }
      p++
    }
    
    if(answer.length>0){
      console.log(answer)
    }else{
      console.log("INVALID RANGE")
    }
  }
  
  function isKaprekar(num){
    if(num ===1){
      return true
    }
    
    let sqNum = (num*num).toString() 
    
    let len = Math.floor(sqNum.length/2)
  
    let num1 = sqNum.slice(0, len)
    let num2 = sqNum.slice(len)
  
    if(Number(num1) + Number(num2) === num){
      return true
    }else{
      return false
    }
  }
  
